contribute/design/proposal-cstor-orchestration.md

```diff
@@ -0,0 +1,311 @@
+           openebs/controller: cstor-controller
+       spec:
+         replicas: 1
+         selector:
+           matchLabels:
+             monitoring: volume_exporter_prometheus
+             openebs/controller: cstor-controller
+             pv: pvc-ee171da3-07d5-11e8-a5be-42010a8001be
+       template:
+         metadata:
+           labels:
+             monitoring: volume_exporter_prometheus
+             openebs/controller: cstor-controller
+             pv: pvc-ee171da3-07d5-11e8-a5be-42010a8001be
+         spec:
+           containers:
+           - name: pvc-ee171da3-07d5-11e8-a5be-42010a8001be-cstor-ctrl
+             image: openebs/cstor:0.7.0               
+             ports:
+             - containerPort: 3260
+               protocol: TCP
+             volumeMounts:
+             - name: cstor-ctrl-istgt-config-map
+               mountPath: /etc/istgt/istgt.conf
+             resources: {}
+           - name: pvc-ee171da3-07d5-11e8-a5be-42010a8001be-cstor-ctrl-exporter
+             image: openebs/m-exporter:0.7.0               
+             ports:
+             - containerPort: 9500
+               protocol: TCP
+             resources: {}             
+           volumes:
+           - name: cstor-ctrl-istgt-config-map
+             configMap:
+               defaultMode: 420
+               name: pvc-ee171da3-07d5-11e8-a5be-42010a8001be-cstor-ctrl-config-map  
+       ```
+       The resources{} will be filled based on the resource (cpu, mem) requests and limits given in the Volume Policies associated with PVC. If nothing has been provided, Kubernetes will assign default values depending on the node resources. Please refer to the [Kubernetes Resource Limits and Request](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#resource-requests-and-limits-of-pod-and-container).
+
+   * maya-apiserver will create the OV cStorReplicas as follows: 
+     * Query for the storage pools matching the given pool name in the PVC and pick up a subset of pools (based on the replica-count of the PVC). 
+     * For each replica, maya-apiserver will create - cStorReplica CRD. This cStorReplica CRD will contain:
+       - cStorPool (unique id like _7b99e406-1260-11e8-aa43-00505684eb2e_)
+       - Unique Name ( an hash will be suffixed to the PVC name like - _pvc-ee171da3-07d5-11e8-a5be-42010a8001be-cstor-rep-9440ab_
+       - Required capacity (obtained from PVC or StorageClass or default value)
+       - OV cStorService IP (obtained from cStorService _(pvc-ee171da3-07d5-11e8-a5be-42010a8001be-cstor-ctrl-service)_ CluserIP )
+       
+       The YAML for the cStorReplica is as follows:
+       ```
+       apiVersion: openebs.io/v1alpha1
+       kind: cStorReplica
+       metadata:
+         name: pvc-ee171da3-07d5-11e8-a5be-42010a8001be-cstor-rep-9440ab
+         poolguid: 7b99e406-1260-11e8-aa43-00505684eb2e
+       spec:
+         cstor-controller-ip: <ip-address>
+         capacity: 5G
```

Comment:
how about adding compression, sync, blocksize, thin provisioning options here, by having defaults for all those?

Reply:
added a note that these customizations will be supported in the future.